### PR TITLE
FIX(Template) typo WantedBy -> Alias

### PR DESCRIPTION
**Pull request by michalxo**

### CODE REVIEW

File: templates/initd.j2 [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#!/bin/sh
+#!/usr/bin/env bash
```
Comment (lenosi) [inline templates/initd.j2]: Bash can't be here. Busybox > ash etc.. it's better to have compatible #!/bin/sh

File: tasks/check.yml [hunk 2]
```diff
@@ -28,3 +28,10 @@
 #     path: /usr/share/upstart
 #   register: upstart_check
 
+
+- name: Set service_opts if undefined
+  when: service_opts is undefined and amq_broker_instance_dir is defined
```
Comment (lenosi) [inline tasks/check.yml]: This is a common ansible role which maybe wants to be used by anyone. I can't approve it during your product dependency, sorry... We had only qdrouterd used in README.md examples.

**Review by lenosi — CHANGES_REQUESTED** [review]

**Comment by fgiorgetti** [comment]
@michalxo @enkeys I believe the new vars are basically adding a similar behavior we already support using the sample below. What do you guys think?

Sample:

ansible-playbook -u root -i localhost, create_service.yml

**create_service.yml**   
- hosts: all
  vars_files:
    - service1.yaml
  roles:
    - role: ansible-service-creator

- hosts: all
  vars_files:
    - service2.yaml
  roles:
    - role: ansible-service-creator

**service1.yaml**
service_opts:
    qdrouterd01:
        Enabled: Yes
        Started: Yes
        ExecStart: "/usr/sbin/qdrouterd -c /var/lib/qdrouterd/qdrouterd.conf"
        User: "qdrouterd"
        Group: "qdrouterd"
        State: "restarted"

**service2.yaml**
service_opts:
    qdrouterd02:
        Enabled: Yes
        Started: Yes
        ExecStart: "/usr/sbin/qdrouterd -c /var/lib/qdrouterd/qdrouterd.conf"
        User: "qdrouterd"
        Group: "qdrouterd"
        State: "restarted"

**Comment by michalxo** [comment]
@fgiorgetti no they don't. They add service which name you already know.
This PR is about adding a service with name, you don't know before playbook-run itself.
That's the difference, much needed for `artemis-<instance-name>.service` creation. ;)

**Comment by fgiorgetti** [comment]
@michalxo Another thing I was able to do here, was defining the service name as below:

**service1.yaml** 
```
---
service_opts:
    "{{ service_name }}":
        Enabled: Yes
        Started: Yes
        ExecStart: "/usr/sbin/qdrouterd -c /var/lib/qdrouterd/qdrouterd.conf"
        User: "qdrouterd"
        Group: "qdrouterd"
        State: "restarted"
```

As you can see there is no "service name" defined, so it is resolved at runtime through `service_name` variable. If that still does not address your issue, maybe you can show me how you're using it so we can evaluate it together.

**Comment by michalxo** [comment]
Here's how I use it from different playbook
```
---
amq_broker_instance_home: "{{ amq_broker_install_dest }}/{{ amq_broker_instance_dir }}"
custom_service_name: "artemis-{{ amq_broker_instance_dir }}"
custom_service_default_service_opts: {
    Description: "{% if 'artemis' in amq_broker_download_url|lower %}Apache ActiveMQ Artemis{% else %}Red Hat AMQ 7{% endif %}",
    After: "syslog.target network.target",
    Type: "simple",
    Environment: [
      "'JAVA_ARGS=\"--add-modules java.xml.bind\"'",
    ],
    PIDFile: "{{ amq_broker_instance_home }}/data/artemis.pid",
    ExecStart: "{{ amq_broker_instance_home }}/bin/artemis run",
    Restart: "no",
    RestartSec: "10",
    User: "{{ amq_broker_user | default(amq_broker_default_user) }}",
    Group: "{{ amq_broker_user | default(amq_broker_default_user) }}",
    UMask: "0027",
    TimeoutStartSec: "3600",
    SuccessExitStatus: "143",
    WantedBy: "multi-user.target",
  }
```

**Comment by fgiorgetti** [comment]
@michalxo I believe, if you have all the variables populated, you can use existing role like demonstrated below. The "service_opts" dictionary will have the service name key using same evaluated value that you have today. See:

```
service_opts:
  "{{ artemis-{{ amq_broker_instance_dir }}":
    Description: "{% if 'artemis' in amq_broker_download_url|lower %}Apache ActiveMQ Artemis{% else %}Red Hat AMQ 7{% endif %}"
    After: "syslog.target network.target"
    Type: "simple"
    Environment: 
      - "'JAVA_ARGS=\"--add-modules java.xml.bind\"'"
    PIDFile: "{{ amq_broker_instance_home }}/data/artemis.pid"
    ExecStart: "{{ amq_broker_instance_home }}/bin/artemis run"
    Restart: "no"
    RestartSec: "10"
    User: "{{ amq_broker_user | default(amq_broker_default_user) }}"
    Group: "{{ amq_broker_user | default(amq_broker_default_user) }}"
    UMask: "0027"
    TimeoutStartSec: "3600"
    SuccessExitStatus: "143"
    WantedBy: "multi-user.target"
```

**Comment by michalxo** [comment]
Yup, you are correct @fgiorgetti , seems to be working :) 
```
TASK [ansible-service-creator : debug] ***************************
ok: [10.37.145.204] => {
    "msg": {
        "artemis-{{ amq_broker_instance_dir }}": {
            "After": "syslog.target network.target",
            "Description": "Apache ActiveMQ Artemis",
            "Environment": [
                "'JAVA_ARGS=\"--add-modules java.xml.bind\"'"
            ],
            "ExecStart": "/home/jamq/ci-artemis/bin/artemis run",
            "Group": "jamq",
            "PIDFile": "/home/jamq/ci-artemis/data/artemis.pid",
            "Restart": "no",
            "RestartSec": "10",
            "SuccessExitStatus": "143",
            "TimeoutStartSec": "3600",
            "Type": "simple",
            "UMask": "0027",
            "User": "jamq",
            "WantedBy": "multi-user.target"
        }
    }
}

TASK [ansible-service-creator : Create systemd service] **********
changed: [10.37.145.204] => (item={'key': 'artemis-ci-artemis', 'value': {'Description': 'Apache ActiveMQ Artemis', 'After': 'syslog.target network.target', 'Type': 'simple', 'Environment': ['\'JAVA_ARGS="--add-modules java.xml.bind"\''], 'PIDFile': '/home/jamq/ci-artemis/data/artemis.pid', 'ExecStart': '/home/jamq/ci-artemis/bin/artemis run', 'Restart': 'no', 'RestartSec': '10', 'User': 'jamq', 'Group': 'jamq', 'UMask': '0027', 'TimeoutStartSec': '3600', 'SuccessExitStatus': '143', 'WantedBy': 'multi-user.target'}})
```
Anyway, I am going to squash this PR and change it only to one minor thing fix.

**Review by lenosi — APPROVED** [review]
Much better! @fgiorgetti Thanks for providing the right example of usage for this case.

**Pull request merged**